stdlib/public/core/String.swift

```diff
@@ -584,24 +584,27 @@ extension String {
 #else
     switch (_core.isASCII, rhs._core.isASCII) {
     case (true, false):
-      let lhsPtr = UnsafePointer<Int8>(_core.startASCII)
-      let rhsPtr = UnsafePointer<UTF16.CodeUnit>(rhs._core.startUTF16)
+      let lhsPtr = unsafeBitCast(_core.startASCII,
+        to: UnsafePointer<CChar>.self)
+      let rhsPtr = UnsafePointer(rhs._core.startUTF16)
 
       return Int(_swift_stdlib_unicode_compare_utf8_utf16(
```

Comment:
If changing these shim functions to accept `unsigned char *` would help, please do.